tests/combine_ecsv_files.py

```diff
@@ -0,0 +1,178 @@
+
+
+def generate_output_file(ecsv_file_list, output_filename, clobber):
+    """Generate combined output ecsv file.
+
+    Parameters
+    ----------
+    ecsv_file_list: list
+        List of ecsv full filenames
+
+    output_filename: string
+         Name of the output combined .ecsv file.
+
+    clobber : Boolean
+        Overwrite existing files with the same name as output_filename?
+
+    Returns
+    -------
+    Nothing.
+    """
+    try:
+        file_start = True
+        n_found = len(ecsv_file_list)
+        filectr = 1
+        for ecsv_filename in ecsv_file_list:
+            table_data = ascii.read(ecsv_filename, format='ecsv')  # Read ecsv file
+            padding = " "*(len(str(n_found))-len(str(filectr)))
+            if len(table_data) < 2:
+                plural_string = ""
+            else:
+                plural_string = "s"
+            print("{}{}/{}: added {} row{} from {}.".format(padding,
+                                                            filectr,
+                                                            n_found,
+                                                            len(table_data),
+                                                            plural_string,
+                                                            ecsv_filename))
+            filectr += 1
+            dataset = os.path.basename(ecsv_filename)[:-5].lower()  # scrape dataset name out of ecsv filename
+            dataset_column = Table.Column(name='datasetName', data=[dataset]*len(table_data))  # make new col
+            table_data.add_column(dataset_column, index=0)  # add dataset column to table data to append.
+
+            if not file_start:  # append out_data with ecsv file data for all files after the list item.
+                out_data = vstack([out_data, table_data])
+            else:  # use the data from the first ecsv file to initialize out_data
+                file_start = False
+                out_data = table_data.copy()
+
+        ascii.write(out_data, output_filename, format='ecsv', overwrite=clobber)  # write output file.
+
+        if n_found == 1:
+            file_plural_string = ""
+        else:
+            file_plural_string = "s"
+        total_rows = len(out_data)  # display total number of rows in output file.
+        if total_rows == 1:
+            row_plural_string = ""
+        else:
+            row_plural_string = "s"
+        print("Wrote {} row{} from {} input file{} to output file {}".format(total_rows,
+                                                                             row_plural_string,
+                                                                             n_found,
+                                                                             file_plural_string,
+                                                                             output_filename))
+    except Exception:
```

Comment:
This code does not add anything to what Python itself would do by default, such as printing out information or parameter values specific to how this code was run, so it does not seem necessary and could be removed.  If you want to keep this handling, I would request that more information be reported by the code on what the code was doing, such as printing out input parameters to help the user better understand what the code was working with when it crashed.

Reply:
Issue resolved by commit [4c49734](https://github.com/spacetelescope/drizzlepac/pull/351/commits/4c49734e37b0473c996cde5f817e567736919522)